src/api/profile.rs

```diff
@@ -18,9 +21,22 @@ pub async fn get_profile(
     mut conn: Connection<Db>,
     id: u32,
 ) -> Result<Json<ApiResponse<Profile>>, GenericError> {
+    let time = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs();
+    let time = time - 86400;
+
     let profile = sqlx::query_as!(Profile, "SELECT * FROM profiles WHERE id = ?", id)
         .fetch_one(&mut **conn)
         .await?;
+    let users = sqlx::query_as!(DBUser, "SELECT * FROM users WHERE id = ?", id)
+        .fetch_one(&mut **conn)
+        .await?;
+
+    if time < users.check_timeout.unwrap() as u64 {
+        println!("not checking (less than 24 hours have passed)")
+    } else {
+        println!("checking");
+        checkdiscordusername(conn, profile.social_discord.clone().unwrap(), id).await;
+    }
```

Comment:
this check should be ran *before* getting the profile - otherwise the username is only going to get updated on the second request

Reply:
please recheck this